doc/Bugsee.md

```diff
@@ -0,0 +1,141 @@
+- [Exception Logging](https://docs.bugsee.com/sdk/flutter/logs/) the app automatically log every unhandled exception: Dart SDK exception are related to data or logic errors and Flutter  SDK errors that are related to layout and rendering issues. The implementation also offer an API to manually log an exception with traces and events.
+- [Video Capturing](https://docs.bugsee.com/sdk/flutter/privacy/video/) video capturing is by default enabled in this project, but it can be turned off using the `videoEnabled` flag in the launchOptions object for Android and iOS.
+- [Log reporting](https://docs.bugsee.com/sdk/flutter/privacy/logs/) all logs are filtered by default using the `_filterBugseeLogs` method, this can be turned off from the app or by removing the call to `setLogFilter` Bugsee method.
+- [Obscure Data](https://docs.bugsee.com/sdk/flutter/privacy/video/#:~:text=Protecting%20flutter%20views): data obscuration is by default enabled in the project in order to protect user-related data from being leaked through captured videos.
+
+**Default configurations:**
+Data obscuration, log collection, log filter and attaching log file features are initialized from the `.env.staging` file.
+```
+BUGSEE_IS_DATA_OBSCURE=true
+BUGSEE_DISABLE_LOG_COLLECTION=true
+BUGSEE_FILTER_LOG_COLLECTION=false
+BUGSEE_ATTACH_LOG_FILE=true
+```
+
+## 2) Implementation
+- [Bugsee Manager](../src/app/lib/business/bugsee/bugsee_manager.dart): a service class that handle the Bugsee intialization, capturing logs, customize Bugsee fields and features (Video capture, data obscure, logs filter...) .
+- [Bugsee Config State](../src/app/lib/business/bugsee/bugsee_config_state.dart): a state class holds all the actual Bugsee features status (whether enabled or not).
+- [Bugsee Repository](../src/app/lib/access/bugsee/bugsee_repository.dart): save and retrieve Bugsee configuration from the shared preference storage.
+- [Bugsee saved configuration](../src/app/lib/access/bugsee/bugsee_configuration_data.dart): holds the Bugsee saved configuration in shared preference, used in [Bugsee Manager](../src/app/lib/business/bugsee/bugsee_manager.dart) to initialize [Bugsee Config State](../src/app/lib/business/bugsee/bugsee_config_state.dart).
+
+### Intecepting exceptions
+Bugsee implementation in this project, by default intecepts all the unhandled Dart and Flutter exception.
+
+`main.dart`
+```dart
+runZonedGuarded(
+    () async {
+      FlutterError.onError =
+          GetIt.I.get<BugseeManager>().inteceptRenderExceptions;
+      await initializeComponents();
+      await registerBugseeManager();
+      runApp(const App());
+    },
+    GetIt.I.get<BugseeManager>().inteceptExceptions,
+  );
+```
+the `inteceptExceptions` and `inteceptRenderExceptions` recerespectively report Dart and Flutter exceptions to the Bugsee Dashboard.
```

Comment:
I think there's a typo in what should be `intercept`

Reply:
and `respectively`

Reply:
I suggest you do a full pass on the text with an AI to proof read it and correct the writing.